packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/bedrock/create-model-customization-job.ts

```diff
@@ -0,0 +1,491 @@
+   * write model artifacts to an S3 bucket.
+   * To pass this role to Amazon Bedrock, the caller of this API must have the iam:PassRole permission.
+   *
+   * @default - use auto generated role
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * The S3 bucket configuration where the training data is stored.
+   *
+   * @see https://docs.aws.amazon.com/bedrock/latest/APIReference/API_TrainingDataConfig.html
+   */
+  readonly trainingData: BucketConfiguration;
+
+  /**
+   * The S3 bucket configuration where the validation data is stored.
+   *
+   * The maximum number is 10.
+   *
+   * @see https://docs.aws.amazon.com/bedrock/latest/APIReference/API_Validator.html
+   */
+  readonly validationData: BucketConfiguration[];
+
+  /**
+   * The VPC configuration.
+   *
+   * @default - no VPC configuration
+   */
+  readonly vpcConfig?: IBedrockCreateModelCustomizationJobVpcConfig;
+}
+
+/**
+ * A Step Functions Task to create model customization in Bedrock.
+ */
+export class BedrockCreateModelCustomizationJob extends sfn.TaskStateBase {
+
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.RUN_JOB,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+  private _role: iam.IRole;
+
+  constructor(scope: Construct, id: string, private readonly props: BedrockCreateModelCustomizationJobProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    this.validateStringLength('clientRequestToken', 1, 256, props.clientRequestToken);
+    this.validatePattern('clientRequestToken', /^[a-zA-Z0-9](-*[a-zA-Z0-9])*$/, props.clientRequestToken);
+    this.validateStringLength('customModelName', 1, 63, props.customModelName);
+    this.validatePattern('customModelName', /^([0-9a-zA-Z][_-]?)+$/, props.customModelName);
+    this.validateArrayLength('customModelTags', 0, 200, props.customModelTags);
+    this.validateStringLength('jobName', 1, 63, props.jobName);
+    this.validatePattern('jobName', /^[a-zA-Z0-9](-*[a-zA-Z0-9\+\-\.])*$/, props.jobName);
+    this.validateArrayLength('jobTags', 0, 200, props.jobTags);
+    this.validateArrayLength('validationData', 1, 10, props.validationData);
+    this.validateArrayLength('securityGroups', 1, 5, props.vpcConfig?.securityGroups);
+    this.validateArrayLength('subnets', 1, 16, props.vpcConfig?.subnets);
+
+    validatePatternSupported(this.integrationPattern, BedrockCreateModelCustomizationJob.SUPPORTED_INTEGRATION_PATTERNS);
+
+    this._role = this.renderBedrockCreateModelCustomizationJobRole();
+    this.taskPolicies = this.renderPolicyStatements();
+
+    if (this.props.customModelKmsKey) {
+      this.props.customModelKmsKey.addToResourcePolicy(new iam.PolicyStatement({
```

Comment:
If we use grant method like below, it will add resource policy to BedrockRole instead of key policy.
In [this document](https://docs.aws.amazon.com/bedrock/latest/userguide/encryption-custom-job.html), I think it would be preferable to implement it according to the documentation. What do you think?